cmake/compileroptions.cmake

```diff
@@ -43,28 +43,20 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     endif ()
 
     set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -Wcast-qual")                # Cast for removing type qualifiers
-    set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -Wno-conversion")            # Implicit conversions that may alter a value
```

Comment:
will this mean we will get sign conversion warnings? I do not think these are implemented properly in gcc/clang/msvc.

Reply:
No, it was unnecessary to disable them. They are not part of `-Wall` or `-Wextra`.

Also the Clang implementation and GCC starting with 10 give saner warnings.

Reply:
> Also the Clang implementation and GCC starting with 10 give saner warnings.

That is interesting! I agree that in theory they have a point.. but well when they are implemented so sloppily there will be too much noise for my taste.

Reply:
My changes will keep the status quo and I will dig into additional GCC warnings at a later date.